contributing/development/core_and_modules/core_types.rst

```diff
@@ -103,7 +103,18 @@ which are equivalent to new, delete, new[] and delete[].
 memnew/memdelete also use a little C++ magic and notify Objects right
 after they are created, and right before they are deleted.
 
-For dynamic memory, use Vector<>.
+For dynamic memory, use Godot's ``Vector<>`` or one of its variations.
+Godot's ``Vector<>`` behaves much like an STL ``std::vector<>``, but is simpler,
```

Comment:
`Vector` is not fully thread safe. CoW copies of `Vector` can access the same data safely through different threads, but several threads cannot access the same instance of one `Vector` safely.